Covenant/Models/Listeners/BridgeListener.cs

```diff
@@ -127,20 +127,35 @@ private async Task RunClient(TcpClient client, CancellationToken token)
             };
             while (!token.IsCancellationRequested)
             {
-                string data = NetworkReadString(stream, token);
-                if (data == null)
+                try
                 {
-                    return;
-                }
-                else
-                {
-                    List<string> parsed = data.ParseExact(((BridgeProfile)this.Profile).WriteFormat.Replace("{DATA}", "{0}").Replace("{GUID}", "{1}")).ToList();
-                    if (parsed.Count == 2)
+                    string data = NetworkReadString(stream, token);
+                    if (data == null)
+                    {
+                        return;
+                    }
+                    else
                     {
-                        _guids.Add(parsed[1]);
-                        await this.InternalListener.Write(parsed[1], parsed[0]);
+                        List<string> parsed = data.ParseExact(((BridgeProfile)this.Profile).WriteFormat.Replace("{DATA}", "{0}").Replace("{GUID}", "{1}")).ToList();
+                        if (parsed.Count == 2)
+                        {
+                            string parsed_guid = parsed[1];
+                            if (parsed_guid.Contains("\0"))
+                            {
+                                // await LoggingService.Log(LogAction.Create, LogLevel.Error, Event);
+                                Console.WriteLine("Grunt GUID too long, removing data that's probably important.");
+                                parsed_guid = parsed_guid.Substring(0, parsed_guid.IndexOf('\0'));
+                            }
+
+                            _guids.Add(parsed_guid);
+                            await this.InternalListener.Write(parsed_guid, parsed[0]);
+                        }
                     }
                 }
+                catch (Exception ex)
+                {
+                    Console.WriteLine("Unkown error, stopping to read from socket");
```

Comment:
I see that Covenant main repository also uses Console.WriteLine, so it's okay.
But let's use here Console.Error.WriteLine.

Reference: https://github.com/cobbr/Covenant/blob/c53155615563cf68979820356b8430e4eb01207d/Covenant/Models/Listeners/InternalListener.cs#L144